pandas/core/arrays/base.py

```diff
@@ -361,23 +362,6 @@ def isna(self) -> ArrayLike:
         """
         raise AbstractMethodError(self)
 
-    def _values_for_argsort(self) -> np.ndarray:
```

Comment:
why can't we leave this and just enhance the signature, passing ``na_position`` ? then the default implementation will just work

Reply:
`_values_for_argsort` is added backed to ExtensionArray interface. Actually, `_values_for_argsort` which calls `np.array` does the same thing as `np.asanyarray` in `nargsort`. But the code in `ExtensionArray.argsort` is organized in more general way.

1. `DatetimeLikeArray` and `PeriodArray`'s own `_values_for_argsort` methods are deleted. Otherwise, NA values in `PeriodArray` would get a negative number from `self._data` which would be placed in the beginning in `nargsort`. 

2. `PandasArray`'s own `_values_for_argsort` is deleted. This method calls `_ndarray` which does the same thing as `np.array` in  its parent class `ExtensionArray._values_for_argsort`.